pinot-controller/src/main/java/com/linkedin/pinot/controller/helix/SegmentStatusChecker.java

```diff
@@ -177,6 +182,13 @@ void updateSegmentMetrics() {
           _metricsRegistry.setValueOfTableGauge(tableName, ControllerGauge.PERCENT_SEGMENTS_AVAILABLE, 100);
           continue;
         }
+
+        if (!idealState.isEnabled()) {
+          LOGGER.warn("Table {} is disabled. Skipping segment status checks", tableName);
```

Comment:
If the table is disabled, the metrics for that table should be removed as well (see https://github.com/linkedin/pinot/pull/2933/files#diff-ec0bdaf9277528f6977db3cd163c518aR278)

Reply:
Thanks for pointing that out. Fixed.